### ISSUE TITLE
Production mode broken for Vaadin + Spring Boot in Vaadin 14.0.9, 14.0.10, 14.0.11

### ISSUE THREAD

**Issue by skremp** [vaadin/flow#6821]
The production mode is broken in case of Vaadin + Spring Boot since 14.0.9.

flow-build-info.json is available with:
{
  "compatibilityMode": false,
  "productionMode": true,
  "enableDevServer": false
}
in /META-INF/VAADIN/config/flow-build-info.json (uber jar)

The project use the same maven setup as in https://vaadin.com/start/latest for Spring Boot.

After starting the Spring Boot UberJar:
```
INFO 16191 --- [           main] c.v.f.s.VaadinServletContextInitializer  : Search for subclasses and classes with annotations took 1 seconds
INFO 16191 --- [           main] c.v.f.server.startup.DevModeInitializer  : Starting dev-mode updaters in XXX folder.
INFO 16191 --- [           main] dev-updater                              : Scanning classes to find frontend configurations and dependencies...
WARN 15370 --- [nio-8090-exec-1] c.v.f.s.DefaultDeploymentConfiguration   : 
====================================================================
Vaadin is running in DEBUG MODE.
In order to run your application in production mode and disable debug features, you should enable it by setting the servlet init parameter productionMode to true.
See https://vaadin.com/docs/v14/flow/production/tutorial-production-mode-basic.html for more information about the production mode.
====================================================================
```

With Vaadin 14.0.8 everything works as expected in production mode.

**Comment by caalador** [comment]
Spring production mode should be added to the application.properties file for jar execution or then the server should be started using `java -Dvaadin.productionMode=true -jar application.jar`

This is due to the fix that prevents us mistakenly trying to use a flow-build-info.json from a add-on jar that mistakenly has been published with the info file. As using a faulty build info with paths caused a lot of exceptions that made no sense.

We are looking into how we could detect the correct file during jar execution.
see vaadin/spring#503 and #6657

**Comment by mehdi-vaadin** [comment]
@skremp This issue should be fixed in Vaadin 14.0.13. So, I'm closing this. Please try 14.0.13 and reopen this issue if you still have the problem.
